src/ol/control/Zoom.js

```diff
@@ -114,6 +114,7 @@ class Zoom extends Control {
       // upon it
       return;
     }
+    view.resolveConstraints(0);
```

Comment:
I'm wondering if a better place for this new line this would be ´ol/View#animate()`, before the last line.

Reply:
I've thought about that as well but didn't manage to get it to work: it was either breaking the "smoothly return view in allowed extent/resolution" effect at the end of an interaction, or causing regressions in other places, depending on where I'd put it.

I agree this does not feel right. To have a cleaner system would mean some rework of the `animate` method IMO, which is a tricky task. The ideal solution would be to get rid of all these `applyTargetState_` and `resolveConstraints` calls and only translate target values into actual view parameters on the main render loop (ans as such apply constraints only once per frame).